iamdata.tf

```diff
@@ -0,0 +1,162 @@
+data "aws_iam_policy_document" "user_self_service_policy" {
+  statement {
+    sid = "ListUsersAccessWithoutMFA"
+
+    actions = [
+      "iam:ListVirtualMFADevices"
+    ]
+
+    effect = "Allow"
+
+    resources = [
+      "*",
+    ]
+  }
+
+  statement {
+    sid = "SelfServiceAccessWithoutMFA"
+
+    actions = [
+        "iam:List*",
+        "iam:GetUser",
+        "iam:GetAccountPasswordPolicy",
+        "iam:ChangePassword",
+        "iam:CreateVirtualMFADevice",
+        "iam:EnableMFADevice"
+    ]
+
+    effect = "Allow"
+
+    resources = [
+      "arn:aws:iam::*:user/$${aws:username}",
+      "arn:aws:iam::*:mfa/$${aws:username}"
+    ]
+  }
+
+  statement {
+    sid = "SelfServiceAccessWithMFA"
+
+    actions = [
+        "iam:Get*",
+        "iam:DeleteSSHPublicKey",
+        "iam:GetSSHPublicKey",
+        "iam:ListSSHPublicKeys",
+        "iam:UpdateSSHPublicKey",
+        "iam:UploadSSHPublicKey",
+        "iam:CreateAccessKey",
+        "iam:DeleteAccessKey",
+        "iam:UpdateAccessKey",
+        "iam:DeleteVirtualMFADevice",
+        "iam:DeactivateMFADevice",
+        "iam:ResyncMFADevice",
+        "iam:UploadSigningCertificate",
+        "iam:UpdateSigningCertificate",
+        "iam:DeleteSigningCertificate",
+        "iam:GenerateServiceLastAccessedDetails"
+    ]
+
+    effect = "Allow"
+
+    resources = [
+      "arn:aws:iam::*:user/$${aws:username}",
+      "arn:aws:iam::*:mfa/$${aws:username}"
+    ]
+
+    condition {
+      test     = "BoolIfExists"
+      variable = "aws:MultiFactorAuthPresent"
+
+      values = [
+        "true"
+      ]
+    }
+  }
+
+  statement {
+    sid = "DenyAllExceptListedIfNoMFA"
+
+    actions = [
+        "iam:ListVirtualMFADevices",
+        "iam:List*",
+        "iam:GetUser",
+        "iam:GetAccountPasswordPolicy",
+        "iam:EnableMFADevice",
+        "iam:CreateVirtualMFADevice",
+        "iam:ChangePassword"
+    ]
+
+    effect = "Deny"
+
+    resources = [
+        "*"
+    ]
+
+    condition {
+      test     = "BoolIfExists"
+      variable = "aws:MultiFactorAuthPresent"
+
+      values = [
+        "false"
+      ]
+    }
+  }
+}
+
+data "aws_iam_policy_document" "role_assumption" {
+  statement {
+    sid = "AssumeRole"
+
+    actions = [
+        "sts:AssumeRole"
+    ]
+
+    effect = "Allow"
+
+    resources = [
+      "${aws_iam_role.this.arn}"
+    ]
+
+    condition {
+      test     = "BoolIfExists"
```

Comment:
Think you want `bool` here
https://github.com/StratusGrid/terraform-aws-iam-group-restricted-admin/pull/2

Reply:
I think one of the bool example under Alternative Combinations should be considered as better option but, according to above documentation, this does emulate the policy that is created when you use the IAM visual editor and choose MFA Required. It also matches the policy that this data source is replacing. I'll approve.